### PR TITLE
improvements for update script

### PR DESCRIPTION
**Pull request by DerDakon**
- add error checking
 - only expire tiles on zoom 0..7 once per day
 - make directory to store OSM data configurable
 - allow updates in arbitrary intervals

### CODE REVIEW

File: import/import.sh [hunk 1]
```diff
@@ -11,20 +11,22 @@ export PATH=/usr/local/bin:/usr/local/sbin:${PATH}
...
-osmdate=`osmconvert planet-latest.osm.pbf --out-timestamp | tr '[TZ]' ' ' | sed 's/ *$//g'`
+osmdate=$(osmconvert ${DATADIR}/planet-latest.osm.pbf --out-timestamp)
 date -u -d "$osmdate" +%s > timestamp
 
 echo "[2/3] Import data into database"
-osmconvert planet-latest.osm.pbf --drop-relations --out-pbf > planet-latest-norelations.osm.pbf
-osm2pgsql --database $DBNAME --username $DBUSER --prefix $DBPREFIX --create --slim --merc --hstore-all --hstore-match-only --hstore-add-index --style openrailwaymap.style --number-processes $NUMPROCESSES --flat-nodes flatnodes --cache $CACHE --input-reader pbf planet-latest-norelations.osm.pbf
-rm planet-latest.osm.pbf
-rm planet-latest-norelations.osm.pbf
+osmconvert ${DATADIR}/planet-latest.osm.pbf --drop-relations --out-pbf > planet-latest-norelations.osm.pbf
```
Comment (rurseekatze) [inline import/import.sh]: I think here is a `${DATADIR}/` missing.

File: import/import.sh [hunk 2]
```diff
@@ -11,20 +11,22 @@ export PATH=/usr/local/bin:/usr/local/sbin:${PATH}
...
+# default to current directory
+DATADIR=${DATADIR:-.}
+
 cd $PROJECTPATH/import
 
 echo "Started processing at $(date)"
 
 echo "[1/3] Downloading planet file"
 wget http://planet.openstreetmap.org/pbf/planet-latest.osm.pbf
-osmdate=`osmconvert planet-latest.osm.pbf --out-timestamp | tr '[TZ]' ' ' | sed 's/ *$//g'`
```
Comment (rurseekatze) [inline import/import.sh]: Does line 24 still work if you remove this here?

File: import/update.sh [hunk 3]
```diff
@@ -5,33 +5,44 @@
...
+if [ ! -s ${DATADIR}/changes.osc ]; then
+	echo "No new data available"
+	exit 0
+fi
+
+osmconvert ${DATADIR}/changes.osc --drop-relations --out-osc > ${DATADIR}/changes-norelation.osc
+rm ${DATADIR}/changes.osc
 
 echo "[2/3] Updating database"
 rm expired_tiles
```
Comment (rurseekatze) [inline import/update.sh]: I think a `${DATADIR}/` is missing here.

File: import/update.sh [hunk 4]
```diff
@@ -5,33 +5,44 @@
...
-osm2pgsql --database $DBNAME --username $DBUSER --prefix $DBPREFIX --append --slim --merc --expire-output expired_tiles --expire-tiles 15 --hstore-all --hstore-match-only --hstore-add-index --style openrailwaymap.style --number-processes $NUMPROCESSES --flat-nodes flatnodes --cache $CACHE changes-norelation.osc
-rm changes-norelation.osc
+osm2pgsql --database $DBNAME --username $DBUSER --prefix $DBPREFIX --append --slim --merc --expire-output expired_tiles --expire-tiles 15 --hstore-all --hstore-match-only --hstore-add-index --style openrailwaymap.style --number-processes $NUMPROCESSES --flat-nodes ${DATADIR}/flatnodes --cache $CACHE ${DATADIR}/changes-norelation.osc
```
Comment (rurseekatze) [inline import/update.sh]: I think a `${DATADIR}/` is missing here.

File: import/update.sh (hunk 3 above)
Comment (DerDakon) [inline import/update.sh]: No, it's your turn to find out why this is actually there and if it should go away.

**Pull request merged**